Add 60k RPM support to voxl-esc driver

modaljc opened the pull request:
## Changes
- Added new packet header for extended RPM range commands
- Added new packet packing function for extended RPM range commands
- Implemented ESC firmware version check at driver start up to determine which type of RPM command to send
- Added voxl-esc driver back to voxl2 build (removing this had removed the voxl-esc section from the Actuators tab in QGC)
- Add Voxl2 IO default parameters

## Testing
- Acro and Manual mode flights were successful
- Successfully spun motors on bench with props attached and verified that RPMs were higher than 32k (./voxl-esc-spin.py --id 0 --rpm 35000 --ramp-time 3.0 --enable-extended-rpm-cmd 1)
- Successfully spun motors on bench with props attached via Actuators tab
- Validated that standard range RPM commands are sent when using old ESC firmware and extended range commands are sent when using new ESC firmware (SW version>=39)

katzfey commented on src/drivers/actuators/voxl_esc/qc_esc_packet.c:
```diff
@@ -119,36 +119,50 @@ int32_t qc_esc_create_pwm_packet4_fb(int16_t pwm0, int16_t pwm1, int16_t pwm2, i
+	
+	// Limit RPMs to prevent overflow when converting to int16_t
+	if (rpm0 > UINT16_MAX) { rpm0 = UINT16_MAX-5; } if (rpm1 > UINT16_MAX) { rpm1 = UINT16_MAX-5; }
+	if (rpm2 > UINT16_MAX) { rpm2 = UINT16_MAX-5; } if (rpm3 > UINT16_MAX) { rpm3 = UINT16_MAX-5; }
```
This is confusing to me. The comment says it is trying to prevent overflow of int16_t. So why would UINT16_MAX-5 accomplish that?

katzfey replied:
Let's let Alex submit his comments to see what his ideas are.

katzfey commented on src/drivers/actuators/voxl_esc/voxl_esc.cpp:
```diff
@@ -1038,6 +1063,12 @@ bool VoxlEsc::updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS],
 
 		} else {
+			if (_extended_rpm) {
+				if (outputs[i] > 65530) outputs[i] = 65530;
```
Perhaps a constant definition of 65530 and 32766 with some comments about why those values were chosen

modaljc replied:
I will make this happen

modaljc commented:
From Alex (referring to qc_esc_packet.c line 154) : 
it is not necessary to cast to doubles when scaling the rpms, integer math is good

I updated this in the latest commit ([3dd7d21](https://github.com/modalai/px4-firmware/commit/3dd7d21ec4cb11f5a16fb83e9fc620a7331adb94)) and removed the double and int16_t casts
